### DISCUSSION TITLE
`infinite moon reader` for Reveal.js slides

### DISCUSSION THREAD

**Discussion by sammo3182** [quarto-dev/quarto-cli#2011] [Q&A] [answered]
I used to built all my slides with Yihui's excellent package `xaringan` and markdown. Recently, I tried to replicate what I set in `xaringan` slides with Quarto. I got most of them done, but still do not know how to achieve `infinite moon reader` for quarto slides. The function (`moon_reader()`) allows users to see the instant changes in the slides without compiling the whole thing all the time. Is there anyway to do that with quarto presentations, or it is just a remark.js thing? (btw, always wonder why Quarto builds slides based on reveal.js not remark.js...)

**Comment by eitsupi** [comment, marked as answer]
Have you tried the `quarto preview` command? (Not a function of R)

> always wonder why Quarto builds slides based on reveal.js not remark.js...

There is a mention of it in the following article, which I think will clear up your question.
https://yihui.org/en/2022/04/quarto-r-markdown/#should-you-switch

**Reply by jjallaire** [reply]
Yes, we aren't using remark.js because its an entirely different markdown rendering stack that's incompatible with Pandoc. There is no realtime preview for Quarto slides however the `quarto preview` method is well tooled in RStudio and VS Code (you need to render to update though).

**Reply by sammo3182** [reply]
@eitsupi Thank you for the answer! I never used this command since I did pretty much everything about Quarto in Rstudio IDE. @jjallaire when you say `quarto preview` tooled in Rstudio, do you mean the preview panel shown after compiling the quarto slides?

**Reply by jjallaire** [reply]
Yes, in the latest version of RStudio (v2022.07) when you Render a Revealjs presentation in will be previewed in a dedicated "Presentation" pane and the preview will sync to the slide being currently edited.

**Reply by sammo3182** [reply]
Great! Thank you! Then this Q&A becomes more like a function request 😝

Hope Quarto could have a realtime preview for Quarto one day; and if someday the quarto preview can preview MS Word, oh my, that, that would be wonderful!! (Sorry for going off-topic, but just can't help. I've dream for this function for a while~)

**Comment by jjallaire** [comment]
If you are using RStudio then MS Word preview should indeed work (you have to have Word installed of course!).
